src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,79 @@
+import json from "./AllowedVocab.json"
+
+/**
+ * Checks, if all words used, are allowed.
+ * 
+ * 
+ * 
+ */
+
+function preprocessingText(text: string){
+    var cleantext = eliminatePunctuation(text);
+    var textarray = transformTextToStringArray(cleantext);
+    return textarray;
+}
+
+export function checkInput(text: string){
+    let preprocessedText : string[] = preprocessingText(text);
+    let issue : string[] = []; //collect all wrong words in a stringarray
+    for (const word of preprocessedText) {
+        if (!json.foo.includes(word)) {//word not included in AllowedVocab.json
+            console.log(word + " ist kein erlaubtes Wort. Bitte verwenden Sie ein anderes!"); //for debugging purposes. The user should get a message on screen telling him what's wrong.
+            issue.push(word);
+        }
+    }
+    let allWrongWords :string = "";
+    allWrongWords += issue.join //Join all wrong words to one String
+    allWrongWords += " sind nicht erlaubte Wörter. Bitte verwenden Sie andere!"
+    console.log(allWrongWords); //Tell the User what's wrong
+    return issue;
+}
+
+/**
+ * Delete any unwanted punctuation
+ */
+
+export function eliminatePunctuation(text: string){
+    var punctuationRegularExpression = /[\u2000-\u206F\u2E00-\u2E7F\\'!"#$%&()*+,\-.\/:;<=>?@\[\]^_`{|}~]/g;//delete some unicoe characters as well as specific punctuation signs.
+    var spaceRegularExpression = /\s+/g;
+    text = text.replace(punctuationRegularExpression, ' ').replace(spaceRegularExpression, ' ');
+    return text;
+}
+
+/**
+ * Collects all positions where there is a whitespace in a string.
+ */
+
+function findSpacePositionsInStringArray(text: string){
+    var index = 0;
+    var spacePositions = [];
```

Comment:
no `var` please. Use `const` instead (or `let`, but only if you need to reassign the variable).